custom/enikshay/integrations/nikshay/repeaters.py

```diff
@@ -186,6 +178,42 @@ def allowed_to_forward(self, test_case):
             return False
 
 
+class NikshayRegisterPrivatePatientRepeater(BaseNikshayRepeater):
+
+    class Meta(object):
+        app_label = 'repeaters'
+
+    include_app_id_param = False
+    friendly_name = _("Forward eNikshay Private Patients to Nikshay (episode case type)")
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.nikshay.views import RegisterNikshayPrivatePatientRepeaterView
+        return reverse(RegisterNikshayPrivatePatientRepeaterView.urlname, args=[domain])
+
+    def allowed_to_forward(self, episode_case):
+        allowed_case_types_and_users = self._allowed_case_type(episode_case) and self._allowed_user(episode_case)
+        if not allowed_case_types_and_users:
+            return False
+
+        try:
+            person_case = get_person_case_from_episode(episode_case.domain, episode_case.get_id)
+        except ENikshayCaseNotFound:
+            return False
+
+        episode_case_properties = episode_case.dynamic_case_properties()
+        return (
+            episode_case_properties.get('nikshay_registered', 'false') == 'false' and
+            not episode_case_properties.get('nikshay_id') and
+            case_properties_changed(episode_case, [PRIVATE_PATIENT_EPISODE_PENDING_REGISTRATION]) and
+            episode_case_properties.get(PRIVATE_PATIENT_EPISODE_PENDING_REGISTRATION, 'yes') == 'no' and
+            is_valid_person_submission(person_case)
+        )
+
+    def send_request(self, payload, url, headers, auth):
+        return simple_xml_post(payload, url, self.operation)
```

Comment:
I noticed you still added `operation` property to the `Repeater` base class; again I'd suggest having a `SOAPRepeaterMixin` (a mixin just being a `class` that you don't intend to instantiate directly, but to supplement a second base class) that has the operation property and this `send_request` method.

(That's because for other repeaters, the `operation` property is meaningless, so it shouldn't go on the base class.)

I would then have `NikshayRegisterPrivatePatientRepeater` subclass both `SOAPRepeaterMixin` and `BaseNikshayRepeater`